src/Util/Http.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Util;
+
+use Sentry\Client;
+use Sentry\Dsn;
+
+/**
+ * @internal
+ */
+final class Http
+{
+    /**
+     * @return string[]
+     */
+    public static function getRequestHeaders(Dsn $dsn, string $sdkIdentifier, string $sdkVersion): array
+    {
+        $headers = [
+            'Content-Type' => 'application/x-sentry-envelope',
+        ];
+
+        $data = [
+            'sentry_version' => Client::PROTOCOL_VERSION,
+            'sentry_client' => $sdkIdentifier . '/' . $sdkVersion,
+            'sentry_key' => $dsn->getPublicKey(),
+        ];
+
+        if (null !== $dsn->getSecretKey()) {
+            $data['sentry_secret'] = $dsn->getSecretKey();
+        }
+
+        $authHeader = [];
+        foreach ($data as $headerKey => $headerValue) {
+            $authHeader[] = $headerKey . '=' . $headerValue;
+        }
+
+        return array_merge($headers, [
+            'X-Sentry-Auth' => 'Sentry ' . implode(', ', $authHeader),
+        ]);
+    }
+
+    /**
+     * @return string[][]
+     */
+    public static function getResponseHeaders(?int $headerSize, string $body): array
```

Comment:
This might be the most fragile part from what I'm reading online. Not saying I have a better option currently but this requires some more scrutiny than I can give at this moment.